File: smartexpense-server/src/main/java/com/smartexpense/server/dto/AuthResponse.java
package com.smartexpense.server.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * Data Transfer Object (DTO) chứa phản hồi xác thực thành công gửi về Client.
 * Chứa token JWT dùng để gán Authorization Header cho các request sau.
 */
@Data
@AllArgsConstructor
public class AuthResponse {
    private String token; // Chuỗi token JWT được sinh ra để định danh phiên làm việc
    private String email; // Email của tài khoản đăng nhập
    private String name;  // Họ tên của tài khoản
    private String role;  // Vai trò của tài khoản (USER/ADMIN)
    private String phone; // Số điện thoại liên kết của tài khoản
}

File: app/src/main/java/com/arijit/budgettracker/api/AuthResponse.kt
package com.arijit.budgettracker.api

/**
 * Lớp dữ liệu (Data Class) nhận thông tin phản hồi từ Server sau khi thực hiện xác thực thành công.
 *
 * @property token Chuỗi JWT Token dùng để xác thực và ủy quyền cho các API tiếp theo.
 * @property email Địa chỉ email của tài khoản.
 * @property name Họ tên đầy đủ của người dùng.
 * @property phone Số điện thoại của người dùng.
 * @property role Vai trò của người dùng trên hệ thống (ADMIN hoặc USER).
 * @property locked Trạng thái khóa tài khoản (1: Đang bị khóa, 0 hoặc null: Đang hoạt động).
 */
data class AuthResponse(
    val token: String,
    val email: String,
    val name: String?,
    val phone: String?,
    val role: String?,
    val locked: Int?
)